### DISCUSSION TITLE
ActionIcons's sizes different than Input or Select sizes

### DISCUSSION THREAD

**Discussion by sowizz** [mantinedev/mantine#3165] [Q&A]
Anyone knows why `ActionIcon`'s sizes differ from other elements? If I want to display `ActionIcon` next to a `Select` and set `size='md'` to both of them I get the following:

![image](https://user-images.githubusercontent.com/1541634/206918808-ce054f5c-a29b-4d5e-bd81-4cbceb31e923.png)

Setting `ActionIcon` size to xl makes it a bit closer, but it's still not ideal:

![image](https://user-images.githubusercontent.com/1541634/206918864-2635377e-a56b-495f-a9b3-29e8f12980b9.png)

Did I chose wrong component to display the filter icon?

**Comment by rtivital** [comment]
ActionIcon is designed to fit into inputs' rightSection, so it is smaller. You can provide your own value in px to match select size: `<ActionIcon size={36} />`

**Reply by sowizz** [reply]
I know I could provide a custom value in px, but I would still prefer to use fixed values like `xs`, `sm`, etc., for consistency sake.

**Comment by farzindev** [comment]
What about moving ActionIcon into Form category and provide a general IconButton component?


right now both of these has `size="md"`
and we should provide a custome `size={42}` to ActionIcon to make it same size as the button with `size="md"`.
![image](https://user-images.githubusercontent.com/6384978/222510577-a989ff2f-9107-4a1e-80bc-6cb578118a9c.png)

**Reply by mattaningram** [reply]
This makes way more sense to have a proper `IconButton` component that matches the `Button` component or have an `IconOnly` prop on `Button`. A lot of the way things are named in Mantine doesn't make a lot of sense to me.

**Comment by mattaningram** [comment]
I'm often using a mix of different Mantine controls together in a single UI like a filter bar or control panel. However it is super frustrating and ugly to use out of the box or even with custom styling because none of their heights match properly. `ActionIcon` has different heights than `Button` which has different heights than `SegmentedControl` (which doesn't even have round pixel values for some reason).

I want all controls at size `md` to have a height of `2.5rem` for example. This takes a lot of complex overrides when it feels like it should just match out of the box. `ActionIcon` is just a `Button` with an `Icon` and no text, it should behave the same way as `Button` in terms of size. If it needs to go inside an input or something then it should have custom styles in that context or just use a smaller `size` prop, not be smaller with the same `size`.

Right now I am trying to add a `SegmentedControl` to my filter row where everything else is set to `2.5rem` height, and for some reason these segmented controls have off-pixel calculated heights. This exact same design could work with same functionality with the `indicator` being sized by its parent (fill height when horizontal) or by its contents so that it has even full pixel values that I can make match my other Mantine controls.

<img width="481" alt="Screen Shot 2023-03-23 at 12 12 16 PM" src="https://user-images.githubusercontent.com/955092/227266942-b746a490-cd1e-4fc5-a51a-ea4f9c2b0f18.png">

This is super frustrating to add custom styles for and makes Mantine forms look sloppy out of the box.
